tests/unit/test_hnswlib.cpp

```diff
@@ -313,14 +314,137 @@ TEST_F(HNSWLibTest, test_hnsw_info) {
+            // M.
+            ASSERT_EQ(infoFiled->fieldType, INFOFIELD_UINT64);
+            ASSERT_EQ(infoFiled->uintegerValue, info.hnswInfo.M);
+        } else if (!strcmp(infoFiled->fieldName, VecSimCommonStrings::HNSW_LEVELS)) {
+            // Levels.
+            ASSERT_EQ(infoFiled->fieldType, INFOFIELD_UINT64);
+            ASSERT_EQ(infoFiled->uintegerValue, info.hnswInfo.levels);
+        } else if (!strcmp(infoFiled->fieldName, VecSimCommonStrings::HNSW_ENTRYPOINT)) {
+            // Entrypoint.
+            ASSERT_EQ(infoFiled->fieldType, INFOFIELD_UINT64);
+            ASSERT_EQ(infoFiled->uintegerValue, info.hnswInfo.entrypoint);
+        } else if (!strcmp(infoFiled->fieldName, VecSimCommonStrings::MEMORY_STRING)) {
+            // Memory.
+            ASSERT_EQ(infoFiled->fieldType, INFOFIELD_UINT64);
+            ASSERT_EQ(infoFiled->uintegerValue, info.hnswInfo.memory);
+        } else {
+            ASSERT_TRUE(false);
+        }
+    }
+}
+
+TEST_F(HNSWLibTest, test_basic_hnsw_info_iterator) {
+    size_t n = 100;
+    size_t d = 128;
+
+    VecSimMetric metrics[3] = {VecSimMetric_Cosine, VecSimMetric_IP, VecSimMetric_L2};
+    for (size_t i = 0; i < 3; i++) {
+        // Build with default args
+        VecSimParams params = {
+            .algo = VecSimAlgo_HNSWLIB,
+            .hnswParams = {
+                .type = VecSimType_FLOAT32, .dim = d, .metric = metrics[i], .initialCapacity = n}};
+        VecSimIndex *index = VecSimIndex_New(&params);
+        VecSimIndexInfo info = VecSimIndex_Info(index);
+        VecSimInfoIterator *infoIter = VecSimIndex_InfoIterator(index);
+        compareHNSWIndexInfoToIterator(info, infoIter);
+        VecSimInfoIterator_Free(infoIter);
+        VecSimIndex_Free(index);
+    }
+}
+
+TEST_F(HNSWLibTest, test_dynamic_hnsw_info_iterator) {
+    size_t n = 100;
+    size_t d = 128;
+    VecSimParams params = {.algo = VecSimAlgo_HNSWLIB,
+                           .hnswParams = {.type = VecSimType_FLOAT32,
+                                          .dim = d,
+                                          .metric = VecSimMetric_L2,
+                                          .initialCapacity = n,
+                                          .M = 100,
+                                          .efConstruction = 250,
+                                          .efRuntime = 400}};
+    float v[d];
+    for (size_t i = 0; i < d; i++) {
+        v[i] = (float)i;
+    }
+    VecSimIndex *index = VecSimIndex_New(&params);
+    VecSimIndexInfo info = VecSimIndex_Info(index);
+    VecSimInfoIterator *infoIter = VecSimIndex_InfoIterator(index);
+    ASSERT_EQ(100, info.hnswInfo.M);
+    ASSERT_EQ(250, info.hnswInfo.efConstruction);
+    ASSERT_EQ(400, info.hnswInfo.efRuntime);
+    ASSERT_EQ(0, info.hnswInfo.indexSize);
+    ASSERT_EQ(-1, info.hnswInfo.levels);
+    ASSERT_EQ(-1, info.hnswInfo.entrypoint);
+    compareHNSWIndexInfoToIterator(info, infoIter);
```

Comment:
Why the ASSERT here, aren't you asserting it in `compareHNSWIndexInfoToIterator`?

Reply:
in this function, you compare the info data to the info iterator data. you don't check the data. This is just sanity.